browser/ui/webui/brave_account/brave_account_ui_desktop.cc

```diff
@@ -33,6 +38,12 @@ class BraveAccountDialogDelegate : public ui::WebDialogDelegate {
     set_dialog_content_url(GURL(kBraveAccountURL));
     set_show_dialog_title(false);
   }
+
+  void OnDialogShown(content::WebUI* webui) override {
+    if (auto* web_contents = webui ? webui->GetWebContents() : nullptr) {
+      BraveAccountWebContentsTag::Mark(web_contents);
```

Comment:
per my other comment it's not clear what `Marked` signifies, but this stays with the webcontents for its entire lifetime once marked?

Reply:
I guess this is a non-tab webcontents so it's lifetime is tied to the brave accounts ui page, but then we should also probably ensure there is no way for it to navigate to any other url, like from an internal link or something? I guess in that case it would still be protected by the standard protections against navigating from http/https -> webui?

Reply:
Basically I'm thinking about someone adding a link to the webui for help/etc... that navigates to http/https and potentially opens some roundabout way to get back to the brave accounts page bypassing the protections here. I think it's safe because of standard chromium protections, but I'm slightly concerned because it's not a regular tab so maybe worth adding a test?

Reply:
`BraveAccountWebContentsTag` is basically a `WebContentsUserData` helper used to mark a given `WebContents` instance as one where the Brave Account `WebUI` is allowed to load. It acts as a runtime flag stored directly on the `WebContents`. This lets the `BraveAccountNavigationThrottle` determine whether it should cancel the navigation or not. This limits loading `brave://account` to within the `ConstrainedWebDialogDelegate` that we show on top of `brave://settings`.

> this stays with the webcontents for its entire lifetime once marked?

Yes.